File: js/tests/howManyVowels.test.js
const countVowels = require("../howManyVowels");

const data = [
  ["Celebration", 5],
  ["Palm", 1],
  ["Prediction", 4],
  ["Suite", 3],
  ["Quote", 3],
  ["Portrait", 3],
  ["Steam", 2],
  ["Tape", 2],
  ["Nightmare", 3],
  ["Convention", 4]
];

describe("countVowels", () => {
  describe("Return count of vowels", () => {
    test.each(data)("countVowels(%p) = %i", (str, result) => {
      let actual = countVowels(str);

      expect(actual).toEqual(result);
    });
  });
});

File: CSharp/Tests/HowManyVowelsTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class HowManyVowelsTest
    {
        [Theory]
        [InlineData("Celebration", 5)]
        [InlineData("Palm", 1)]
        [InlineData("Prediction", 4)]
        [InlineData("Suite", 3)]
        [InlineData("Quote", 3)]
        [InlineData("Portrait", 3)]
        [InlineData("Steam", 2)]
        [InlineData("Tape", 2)]
        [InlineData("Nightmare", 3)]
        [InlineData("Convention", 4)]
        public void CountVowels_StringInput_ReturnCountOfVowels(string str, int expected)
        {
            var actual = HowManyVowels.CountVowels(str);

            Assert.Equal(expected, actual);
        }
    }
}